test/e2e/flowaggregator_test.go

```diff
@@ -259,18 +266,49 @@ func checkRecordsForFlows(t *testing.T, data *TestData, srcIP string, dstIP stri
 						t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
 					}
 				}
-				// Check the bandwidth using octetDeltaCount in data record.
 				if checkBandwidth {
-					checkBandwidthFromRecord(t, record, bandwidth)
+					exportTime := int64(getUnit64FieldFromRecord(t, record, "flowEndSeconds"))
+					if exportTime <= timeStartSec+iperfTime-aggregatorExportInterval && exportTime > timeStartSec+aggregatorExportInterval {
+						curOctetDeltaCount := getUnit64FieldFromRecord(t, record, "octetDeltaCountFromSourceNode")
+						t.Logf("Check the bandwidth using octetDeltaCountFromSourceNode in data record.")
+						checkBandwidthByInterval(t, bandwidth, curOctetDeltaCount, float64(aggregatorExportInterval), "octetDeltaCountFromSourceNode")
+					}
+					curOctetTotalCount := getUnit64FieldFromRecord(t, record, "octetTotalCountFromSourceNode")
+					if curOctetTotalCount > octetTotalCount {
+						octetTotalCount = curOctetTotalCount
+					}
 				}
 			}
 		}
 	}
+	if checkBandwidth && octetTotalCount > 0 {
+		t.Logf("Check the average bandwidth using octetTotalCountFromSourceNode in data record.")
+		checkBandwidthByInterval(t, bandwidth, octetTotalCount, float64(iperfTime), "octetTotalCountFromSourceNode")
```

Comment:
can these lines be added to the above if block to simplify the code?

Moreover, I presume you are maintaining octetTotalCount
to get the max till now--wondering what is the use of tracking this? With aggregated records, isn't the field "octetTotalCountFromSourceNode"
always increasing?  Do you see records that belong to one flow not following any trend?

Reply:
As 'checkBandwidth` is a common check, these lines can be moved to the above if-block. Please add the reason why do you have to consider the max of `octetTotalCount` in a comment.

Reply:
If this check is being done after iterating through all the records, please mention that in a comment as all the checks are done for every record.